src/octoprint/plugins/softwareupdate/__init__.py

```diff
@@ -1492,6 +1503,17 @@ def get_wizard_details(self):
     def on_event(self, event, payload):
         from octoprint.events import Events
 
+        if (
+            event == Events.PRINT_DONE
+            or event == Events.PRINT_FAILED
+            and self._queued_updates is not None
+        ):
+            self.perform_updates(
+                targets=self._queued_updates["targets"],
+                force=self._queued_updates["force"],
+            )
+            self._queued_updates = None
+
```

Comment:
I'm leaning towards not immediately starting the update but having a countdown here to allow the user to still intervene.

Reply:
added timer, but still want to re-do the way the PNotify is getting made and utilize the _showPopup function.